File: ios/Classes/HLSPlayer/HMSHLSCueExtension.swift
//
//  HMSHLSCueExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 24/05/23.
//

import Foundation
import HMSHLSPlayerSDK

class HMSHLSCueExtension{
    
    static func toDictionary(_ hmsHlsCue:HMSHLSCue)-> [String: Any?]{
        
        var dict = [String: Any?]()
        
        dict["id"] = hmsHlsCue.id
        dict["start_date"] = "\(hmsHlsCue.startDate)"
        if let endDate = hmsHlsCue.endDate{
            dict["end_date"] = "\(endDate)"
        }
        dict["payload"] = hmsHlsCue.payload
        
        return dict
    }
}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/hls_player/HMSHLSCueExtension.kt
package live.hms.hmssdk_flutter.hls_player

import live.hms.hls_player.HmsHlsCue
import java.text.SimpleDateFormat

class HMSHLSCueExtension {
    companion object {
        fun toDictionary(hmsHlsCue: HmsHlsCue): HashMap<String, Any?> {
            val args = HashMap<String, Any?>()

            args["id"] = hmsHlsCue.id
            args["start_date"] = hmsHlsCue.startDate
            args["end_date"] = hmsHlsCue.endDate
            args["payload"] = hmsHlsCue.payloadval

            return args
        }
    }
}
